app/logic/Contacts/ActiveSync/ActiveSyncAddressbook.ts

```diff
@@ -1,16 +1,129 @@
 import { Addressbook } from "../Addressbook";
+import { SQLAddressbook } from '../SQL/SQLAddressbook';
+import { SQLPerson } from '../SQL/SQLPerson';
 import { ActiveSyncPerson } from "./ActiveSyncPerson";
-import { ActiveSyncGroup } from "./ActiveSyncGroup";
-import type { ActiveSyncAccount } from "../../Mail/ActiveSync/ActiveSyncAccount";
+import { EASError, type ActiveSyncAccount, type ActiveSyncPingable } from "../../Mail/ActiveSync/ActiveSyncAccount";
+import { kMaxCount, ensureArray } from "../../Mail/ActiveSync/ActiveSyncFolder";
+import { NotSupported } from "../../util/util";
+import type { ArrayColl } from "svelte-collections";
 
-export class ActiveSyncAddressbook extends Addressbook {
+export class ActiveSyncAddressbook extends Addressbook implements ActiveSyncPingable {
   readonly protocol: string = "addressbook-activesync";
+  readonly persons: ArrayColl<ActiveSyncPerson>;
   account: ActiveSyncAccount;
+  syncKeyBusy: Promise<any> | null;
+  readonly folderClass = "Contacts";
+
+  get serverID() {
+    return new URL(this.url).searchParams.get("serverID");
+  }
+
+  async ping() {
+    await this.listContacts();
+  }
 
   newPerson(): ActiveSyncPerson {
     return new ActiveSyncPerson(this);
   }
-  newGroup(): ActiveSyncGroup {
-    return new ActiveSyncGroup(this);
+  newGroup(): never {
+    throw new NotSupported("ActiveSync does not support distribution lists");
+  }
+
+  async queuedSyncRequest(data: any): Promise<any> {
+    if (!this.syncState && !this.syncKeyBusy) try {
+      // First request must be an empty request.
+      this.syncKeyBusy = this.makeSyncRequest();
+      await this.syncKeyBusy;
+    } finally {
+      this.syncKeyBusy = null;
+    }
+    while (this.syncKeyBusy) try {
+      await this.syncKeyBusy;
+    } catch (ex) {
+      // If the function currently holding the sync key throws, we don't care.
+    }
+    try {
+      this.syncKeyBusy = this.makeSyncRequest(data);
+      return await this.syncKeyBusy;
+    } finally {
+      this.syncKeyBusy = null;
+    }
+  }
+
+  protected async makeSyncRequest(data?: any): Promise<any> {
```

Comment:
Can we please document the return value, at least in a JSDoc comment?

Also, it's kind-of funny to have an `async syncReq()` ;-) Better: `makeSynchronizeRequest()`. Ditto for `queued` above.

Reply:
The return value depends on the request being made, so no, I can't document it.

And it's called `makeSyncRequest` because the name of the request that's being made is "Sync".

Reply:
> The return value depends on the request being made, so no, I can't document it.

You're returning `response.Collections.Collection`, so there must be something that you can say about the result. "It's an array of ..."

> And it's called `makeSyncRequest` because the name of the request that's being made is "Sync".

Yes, I understood (after a while) what you meant, that's why I made a concrere suggestion. 

I thought at first that you make a non-async request. So, "Sync" is ambiguous.